mozci/push.py

```diff
@@ -1002,14 +1011,35 @@ def generate_all_shadow_scheduler_tasks(
             would have scheduled, or an exception instance in case the shadow
             scheduler failed.
         """
-        names = [
-            label.split("shadow-scheduler-")[1]
-            for label in self.scheduled_task_labels
-            if "shadow-scheduler" in label
-        ]
-        for name in sorted(names):
+        for name in sorted(self.shadow_scheduler_names):
             try:
-                yield name, self.get_shadow_scheduler_tasks(name)
+                yield name, set(
+                    t["label"] for t in self.get_shadow_scheduler_tasks(name)
+                )
+            except Exception as e:
+                yield name, e
+
+    def generate_all_shadow_scheduler_config_groups(
```

Comment:
Won't force you to block on it, but would be nice to have a test similar to the task version:
https://github.com/mozilla/mozci/blob/master/tests/test_push.py#L361

Reply:
Done!